modules/adpod.js

```diff
@@ -0,0 +1,364 @@
+ * @param {Function} fn reference to original function (used by hook logic)
+ * @param {Object} bid incoming bid object
+ * @param {Object} bidRequest bidRequest object of associated bid
+ * @param {Object} videoMediaType copy of the `bidRequest.mediaTypes.video` object; used in original function
+ * @param {String} context value of the `bidRequest.mediaTypes.video.context` field; used in original function
+ * @returns {boolean} this return is only used for adpod bids
+ */
+export function checkVideoBidSetupHook(fn, bid, bidRequest, videoMediaType, context) {
+  if (context === ADPOD) {
+    if (!utils.deepAccess(bid, 'meta.iabSubCatId')) {
+      return false;
+    }
+
+    if (utils.deepAccess(bid, 'video')) {
+      if (!utils.deepAccess(bid, 'video.context') || bid.video.context !== ADPOD) {
+        return false;
+      }
+
+      if (!utils.deepAccess(bid, 'video.durationSeconds') || bid.video.durationSeconds <= 0) {
+        return false;
+      } else {
+        let isBidGood = checkBidDuration(bidRequest, bid);
+        if (!isBidGood) return false;
+      }
+    }
+
+    if (!config.getConfig('cache.url') && bid.vastXml && !bid.vastUrl) {
+      utils.logError(`
+        This bid contains only vastXml and will not work when a prebid cache url is not specified.
+        Try enabling prebid cache with pbjs.setConfig({ cache: {url: "..."} });
+      `);
+      return false;
+    };
+
+    return true;
+  } else {
+    fn.call(this, bid, bidRequest, videoMediaType, context);
+  }
+}
+
+/**
+ * This function reads the (optional) settings for the adpod as set from the setConfig()
+ * @param {Object} config contains the config settings for adpod module
+ */
+export function adpodSetConfig(config) {
+  if (config.bidQueueTimeDelay !== undefined) {
+    if (typeof config.bidQueueTimeDelay === 'number' && config.bidQueueTimeDelay > 0) {
+    // add check to see if config setting is too high?
+      queueTimeDelay = config.bidQueueTimeDelay;
+    } else {
+      utils.logWarn(`Detected invalid value for adpod.bidQueueTimeDelay in setConfig; must be a positive number.  Using default: ${queueTimeDelay}`)
+    }
+  }
+
+  if (config.bidQueueSizeLimit !== undefined) {
+    if (typeof config.bidQueueSizeLimit === 'number' && config.bidQueueSizeLimit > 0) {
+    // add check to see if config setting is too high? too low?
+      queueSizeLimit = config.bidQueueSizeLimit;
+    } else {
+      utils.logWarn(`Detected invalid value for adpod.bidQueueSizeLimit in setConfig; must be a positive number.  Using default: ${queueSizeLimit}`)
+    }
+  }
+}
+config.getConfig('adpod', config => adpodSetConfig(config.adpod));
+
+/**
+ * This function initializes the adpod module's hooks.  This is called by the corresponding adserver video module.
+ */
+export function initAdpodHooks() {
+  function setupHookFnOnce(hookId, hookFn, priority = 15) {
+    let result = hooks[hookId].getHooks({hook: hookFn});
```

Comment:
So they would be added to the `hooks['checkAdUnitSetup'] object, but they wouldn't show when I execute the `getHooks()` call because I'm effectively filtering that `get` to only look for the function that's inside this module.

I tested this out locally and even when I name the hook function the same (ie `checkAdUnitSetupHook`) - the hooked function won't return with this type of call because the function signature is different.